tutorials/javaFxPart5.md

```diff
@@ -0,0 +1,419 @@
+within the stylesheet.
+
+**main.css**
+```css
+.root {
+    duke: rgb(237, 255, 242);     /* Create a looked-up color called "duke" within root */
+    -fx-background-color: duke;
+}
+
+.scroll-bar {
+    ...
+    -fx-background-color: duke;   /* Using looked-up color */
+}
+```
+
+### Gradient color
+
+To set a [gradient color](https://docs.oracle.com/javafx/2/api/javafx/scene/doc-files/cssref.html#typepaint),
+we can either use linear or radial gradient.
+
+```css
+-fx-background-color: linear-gradient(to bottom right, red, blue);
+-fx-background-color: radial-gradient(radius 100%, red, purple, blue);
+```
+
+
+## States
+
+You can set different properties for different states of a node.
+
+For example, let us take a look at the send button:
+
+**main.css**
+```css
+.button {
+    -fx-background-color:springgreen;
+    -fx-font: italic bold 16px "Arial";
+}
+
+.button:hover {
+    -fx-background-color:cyan;
+    -fx-font-size: 18px;
+}
+
+.button:pressed {
+    -fx-background-color:orange;
+    -fx-font-size: 20px;
+}
+```
+
+In this case, there are three different states:
+-  Default, when nothing happens to the button
+- `:hover`, when your mouse hovers the button
+- `:pressed`, when you click on the button
+
+![Button States Demo](images/javafx/ButtonStatesDemo.gif)
+
+This works for TextFields/Labels/Images as well.
+
+Other state(s) include:
+
+- `:focused`, when the node is selected (e.g., selecting the TextField)
+
+
+## Image and shadows
+
+Since an ImageView is a [node](https://docs.oracle.com/javafx/2/api/javafx/scene/doc-files/cssref.html#node), we can
+play around with the size, opacity, rotation, and more.
+
+(this also means that the following properties can also be used for other kind of nodes).
+
+
+```css
+#displayPicture {
+    /* Shadow effect on image. */
+    -fx-effect: dropshadow(gaussian, rgba(0, 0, 0, 0.2), 10, 0.5, 5, 5);
+
+    /* Change size of image. */
+    -fx-scale-x: 1;
+    -fx-scale-y: 1;
+
+    /* Rotate image clockwise by degrees. */
+    -fx-rotate: 0;
+}
+```
+
+
+<box type="info" seamless>
+
+Note that the `#` symbol is used to select the node by its ID (or variable),
+meaning it applies the style to just `displayPicture` and not to `ImageView`s in general.
+This is different from using a class selector (which uses a `.`).
+</box>
+
+There are 2 different types of [shadow effects](https://docs.oracle.com/javafx/2/api/javafx/scene/doc-files/cssref.html#typeeffect):
+
+```css
+    -fx-effect: dropshadow(gaussian, rgba(0, 0, 0, 0.2), 10, 0.5, 5, 5);
+    -fx-effect: innershadow(gaussian, rgba(0, 0, 0, 0.2), 10, 0.5, 5, 5);
+```
+![Types of Shadows demo](images/javafx/TypesOfShadows.png)
+
+
+If you use an image with a transparent background (e.g., [this doggy](https://pixabay.com/illustrations/dachshund-dog-hot-dog-animals-6336468/)),
+the shadow effect will not apply to the transparent areas.
+
+![Transparent background image demo](images/javafx/ImageShadowDemo.png)
+
```

Comment:
Using speech bubble as a caption might cause confusion. Perhaps crop it out and use something like this instead?

```suggestion
Drop shadow | Inner shadow | Transparent background
------------|--------------|-----------------------
![](images/javafx/ImageShadowDemo.png)| ![](images/javafx/ImageShadowDemo.png) | ![](images/javafx/ImageShadowDemo.png)
```suggestion

```